components/Header.tsx

```diff
@@ -1,17 +1,51 @@
-import { Flex, Heading } from "@chakra-ui/react";
+import {
+  Box,
+  Button,
+  Container,
+  Flex,
+  Heading,
+  useDisclosure,
+} from "@chakra-ui/react";
+import { useLogout, useUserQuery } from "@lib/hooks/user";
 import { Colors, useColor } from "@styles/colors";
+import { useState } from "react";
+import { v4 as uuid } from "uuid";
+import Login from "./Login";
 
 export default function Header() {
+  const { isAuthenticated } = useUserQuery();
+  const { isOpen, onClose, onOpen } = useDisclosure();
+
+  const [loginModalKey, setLoginModalKey] = useState("");
+  const handleLoginClick = () => {
+    // Reset login modal each time it is opened
+    setLoginModalKey(uuid());
+    onOpen();
+  };
+
+  const logout = useLogout();
+
   return (
-    <Flex
-      backgroundColor={useColor(Colors.secondaryBackground)}
-      padding={4}
-      alignItems="center"
-      justifyContent="center"
-    >
-      <Heading as="h1" size="md">
-        Book Bazar
-      </Heading>
-    </Flex>
+    <Box as="header" backgroundColor={useColor(Colors.secondaryBackground)}>
+      <Container py={5} maxWidth="container.md">
+        <Flex alignItems="baseline" justifyContent="space-between">
+          <Heading as="h1" size="md">
+            Book Bazar
+          </Heading>
+          <div />
+          {!isAuthenticated ? (
+            <Button variant="link" onClick={handleLoginClick}>
+              Login
+            </Button>
+          ) : (
+            <Button variant="link" onClick={() => logout()}>
```

Comment:
Unless I'm missing something, you could just pass `logout` directly

Reply:
Ah this is just a convention that comes from calling JavaScript functions. Say `logout` was modified to be of type `(shouldReload: boolean) => Void`. Then if we passed the function directly, we would be implicitly writing `onClick={(clickEvent) => logout(clickEvent)}` which is equivalent to `onClick={() => logout(true)}`, without realizing it.

Since we're using TypeScript it should avoid this issue but it's just a practice I picked up from when not using it. My preference is to only directly pass functions that are defined in the component such as 
```typescript
const handleLogoutClick = () => { ... };
<Button variant="link" onClick={handleLogoutClick}>
```

If you feel strongly about not doing this though, I can change it.

Reply:
Just realized this is actually one of the cases where we can't pass `logout` directly for the reason I mentioned.

Reply:
Okay. Makes sense to me why you are doing it this way. I'll make sure to do this in my code as well. Sorry for the bother.

Reply:
No bother! Happy to share